File: packages/deepeye-backend/app/api/v1/nodes.py
"""Node API routes."""

from fastapi import APIRouter, HTTPException, status

from app.models.schemas.node import (
    NodeInfo,
    NodeListResponse,
    NodeExecutionRequest,
    NodeExecutionResult,
)
from app.services.node_service import NodeService

router = APIRouter(prefix="/nodes", tags=["nodes"])

node_service = NodeService()


@router.get("", response_model=NodeListResponse)
async def list_nodes():
    """Get all registered nodes."""
    return node_service.get_all_nodes()


@router.get("/{node_type}", response_model=NodeInfo)
async def get_node_info(node_type: str):
    """Get detailed information about a node."""
    node_info = node_service.get_node_info(node_type)
    if not node_info:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"Node type '{node_type}' not found",
        )
    return node_info


@router.post("/{node_type}/execute", response_model=NodeExecutionResult)
async def execute_node(node_type: str, request: NodeExecutionRequest):
    """Execute a single node."""
    result = await node_service.execute_node(
        node_type=node_type,
        inputs=request.inputs,
        config=request.config,
    )
    
    if result.status == "failed":
        raise HTTPException(
            status_code=status.HTTP_400_BAD_REQUEST,
            detail=result.error or "Node execution failed",
        )
    
    return result

File: packages/deepeye-frontend/src/shared/api/nodes.ts
/**
 * 节点 API
 * 
 * 对接后端节点接口：
 * - GET /api/v1/nodes - 列出所有节点
 * - GET /api/v1/nodes/{type} - 获取节点信息
 * - POST /api/v1/nodes/{type}/execute - 执行单个节点
 */

import { apiClient } from './client'

// ============ 类型定义 ============

export interface PortInfo {
  name: string
  type: string
  description?: string
  required?: boolean
}

export interface NodeInfo {
  type: string
  label: string
  category: string
  description?: string
  inputs: PortInfo[]
  outputs: PortInfo[]
  config_schema?: Record<string, any>
}

export interface NodeListResponse {
  total: number
  nodes: NodeInfo[]
}

export interface NodeExecutionRequest {
  inputs: Record<string, any>
  config?: Record<string, any>
}

export interface NodeExecutionResult {
  status: 'success' | 'failed'
  outputs: Record<string, any>
  execution_time: number
  error: string | null
}

// ============ API 函数 ============

export const nodesAPI = {
  list: () => apiClient.get<NodeListResponse>('/nodes'),
  getInfo: (type: string) => apiClient.get<NodeInfo>(`/nodes/${type}`),

  execute: (type: string, inputs: Record<string, any>, config?: Record<string, any>) =>
    apiClient.post<NodeExecutionResult>(`/nodes/${type}/execute`, {
      inputs,
      config: config || {},
    }),
}

